osu.Game.Rulesets.Catch/Edit/Blueprints/JuiceStreamSelectionBlueprint.cs

```diff
@@ -59,10 +95,38 @@ protected override void Update()
             pathCache.Validate();
         }
 
+        protected override void OnSelected()
+        {
+            initializeJuiceStreamPath();
+            base.OnSelected();
+        }
+
+        protected override bool OnMouseDown(MouseDownEvent e)
+        {
+            if (!IsSelected) return base.OnMouseDown(e);
+
+            switch (e.Button)
+            {
+                case MouseButton.Left when e.ControlPressed:
+                    editablePath.AddVertex(editablePath.ToRelativePosition(e.ScreenSpaceMouseDownPosition));
+                    return true;
+
+                case MouseButton.Right:
+                    // Record the mouse position to be used in the "add vertex" action.
+                    rightMouseDownPosition = editablePath.ToRelativePosition(e.ScreenSpaceMouseDownPosition);
+                    break;
+            }
+
+            return base.OnMouseDown(e);
+        }
+
         private void onDefaultsApplied(HitObject _)
         {
             computeObjectBounds();
             pathCache.Invalidate();
+
+            if (lastSliderPathVersion != HitObject.Path.Version.Value)
+                initializeJuiceStreamPath();
```

Comment:
Maybe not a huge issue, but this can potentially get run when the blueprint is not selected. Not sure if that's the intention, or whether it needs to be guarded against.

One example of triggering this is to drag a juice stream's end node on the timeline to create a repeat (without selecting it first). I think other similar scenarios can be found with systems like undo/redo as well.

Reply:
It is not necessary (if the code is changed to make `EditablePath` initialized when the blueprint is selected) but doesn't hurt either way. Probably the current way is simpler.

Reply:
Okay. I guess we'll leave it until it causes any performance concern or otherwise.